uniffi_bindgen/src/bindings/kotlin/gen_kotlin/mod.rs

```diff
@@ -510,6 +511,29 @@ mod filters {
         Ok(as_ct.as_codetype().literal(literal, ci))
     }
 
+    // Get the idiomatic Kotlin rendering of an individual enum variant's discriminant
+    pub fn variant_discr_literal(e: &Enum, index: &usize) -> Result<String, askama::Error> {
+        let literal = e.variant_discr(*index).expect("invalid index");
+        match literal {
+            // Kotlin doesn't convert between signed and unsigned by default
+            // so we'll need to make sure we define the type as appropriately
+            LiteralMetadata::UInt(v, _, _) => {
+                if let Some(t) = e.variant_discr_type() {
+                    match t {
+                        Type::Int8 | Type::Int16 | Type::Int32 | Type::Int64 => Ok(v.to_string()),
+                        Type::UInt8 | Type::UInt16 | Type::UInt32 | Type::UInt64 => {
+                            Ok(v.to_string() + "u")
+                        }
+                        _ => unreachable!("only unsigned is supported at this time"),
+                    }
+                } else {
+                    unreachable!("enum hasn't defined a repr")
+                }
+            }
+            _ => unreachable!("expected an UInt!"),
```

Comment:
If you made [this struct](https://github.com/mozilla/uniffi-rs/blob/main/uniffi_bindgen/src/backend/filters.rs#L16) pub and imported it, you could maybe change these to `return askama::Error::Custom(Box::new(UniFFIError::new("mesage...")))`

(this and the one unreachable above - the first unreachable here is fine but maybe with a different message as I wrote above)